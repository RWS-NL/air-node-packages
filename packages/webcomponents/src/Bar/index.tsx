import { Box } from '@mui/material';
import clsx from 'clsx';
import React, { memo, PropsWithChildren } from 'react';
import css from './Bar.scss';

interface BarProps {
  /** Additional CSS classes to apply to the Bar component */
  classes?: string | string[];
}

/**
 * Constructs a horizontal Bar section using pre-defined Rijkswaterstaat styling
 * @example
 * ```jsx
 * <Bar>
 *   <div> Some Children </div>
 * </Bar>
 * ```
 */
export const Bar = memo(({ classes, children, ...props }: PropsWithChildren<BarProps>) => (
  <Box {...props} component='div' className={clsx(css.bar, classes)}>
    {children}
  </Box>
));
