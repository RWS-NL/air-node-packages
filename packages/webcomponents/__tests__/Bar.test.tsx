import { Bar } from '@src/Bar';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

const children = <p>child</p>;
let bar: ShallowWrapper;

beforeAll(() => {
  bar = shallow(<Bar classes='blue-bar'>{children}</Bar>);
});

test('should render as expected', () => {
  expect(bar).toMatchSnapshot();
});

test('bar has a child component', () => {
  expect(bar.children()).toHaveLength(1);
  expect(bar.childAt(0).text()).toStrictEqual('child');
});

test('bar takes the classes from props', () => {
  expect(bar.hasClass('blue-bar')).toBe(true);
});
