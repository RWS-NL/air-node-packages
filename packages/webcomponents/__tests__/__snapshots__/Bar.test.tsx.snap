// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<ForwardRef(Box)
  className="bar blue-bar"
  component="div"
>
  <p>
    child
  </p>
</ForwardRef(Box)>
`;
